### DISCUSSION TITLE
Selecting stops for Tutti

### DISCUSSION THREAD

**Discussion by Bumblebee001** [GrandOrgue/GoOdf#61] [General]
Is there a situation when a selected stop or coupler would be off?

I see no reason for having aselected element to be off by default!

What am I missing?

**Comment by larspalo** [comment]
A "Tutti" is still implemented as a general as there's no other dedicated element for it. In an odf general/divisional a stop or coupler (or any other element) should be off if it is needed to be set as off, and it could very well be intended to be (to make sure some elements are de-activated when the combination is pushed). Remember that the odf based generals (and divisionals) were created way before the built in versions existed in GrandOrgue. The expected behaviour way back then was mostly to just add the generals and/or divisionals empty but with all available (relevant) elements for the user to set at will. If it's not created as protected, the user can change the content of the combination!

Nowadays, the odf generals/divisionals should mostly be used for "fixed" combinations or "tricks" and may very well not at all include all available stops/couplers or other elements. Still, the decision if a certain available element should be on or off is not obvious - also, if an element is not available, think what would happen when that combination would be pressed: an active element won't be "cancelled" by the current combination resulting in another combination than was perhaps intended! But it makes more sense now to create odf combinations as protected so that they are really "fixed" and cannot be tampered with by the user.

The choice if an element just added to a general or divisional in GOODF should be set as on or off by default is an open question. I've implemented it as off by default because that was how I would write them into the odf earlier (to be filled by the user)... We have a slightly different situation today though. I've used them too little to really have a strong opinion if it would indeed be better to have the just added elements as on by default instead of off.

The main point is that odf generals could be used for many other things than just to create a "Tutti" combination.
